Initial commit for the second homework

daria-proleskovska opened the pull request.

daria-proleskovska committed:
Initial commit for the second homework

myroslav-kolomiiets commented on hw2/index.js:
```diff
@@ -0,0 +1,87 @@
+ * Task 1
+ */
+
+function durationBetweenDates(dateStart = '24 Feb 2022', dateEnd = new Date(), dimension = 'days') {
```
Тут плутанина з типами даних. dateStart буде string, dateEnd - object. Все працює і так, але рекомендую зробити рішення, в якому типи початкових дат, кінцевих дат, і ті що передаються при виклику і дефолтні, мали однаковий тип даних.

myroslav-kolomiiets commented on hw2/index.js:
```diff
@@ -0,0 +1,87 @@
+
+function optimizer(data) {
+    return Object.fromEntries(
+        Object.entries(data).map(([key, value]) => [key.toLowerCase(), parseFloat(value).toFixed(2)])
```
👍

myroslav-kolomiiets commented on hw2/index.js:
```diff
@@ -0,0 +1,87 @@
+        return number;
+    } else if (number % 2 === 0) {
+        // even number
+        return number - 1 + recursiveOddSumTo(number - 2);
```
```suggestion
        return recursiveOddSumTo(number - 1);
```
Трохи перемудрили)

myroslav-kolomiiets commented on hw2/index.js:
```diff
@@ -0,0 +1,87 @@
+ */
+
+function recursiveOddSumTo(number) {
+    if (number === 1 || number === 0) {
```
```suggestion
    if (number === 1) {
```
Достатньо тільки цієї перевірки

myroslav-kolomiiets commented on hw2/index.js:
```diff
@@ -0,0 +1,87 @@
+    } else if (number % 2 === 0) {
+        // even number
+        return number - 1 + recursiveOddSumTo(number - 2);
+    } else {
+        // odd number
+        return number + recursiveOddSumTo(number - 2);
+    }
```
```suggestion
    }
    // odd number
    return number + recursiveOddSumTo(number - 2);
```
Теж саме, тільки лаконічніше

myroslav-kolomiiets commented on hw2/index.js:
```diff
@@ -0,0 +1,87 @@
+ * Task 4
+ */
+
+function iterativeOddSumTo(number) {
```
👍

daria-proleskovska committed:
Improvements

- task 1: change default dateEnd type to string
- task 3: improve function logic

myroslav-kolomiiets commented on hw2/index.js:
```diff
@@ -0,0 +1,88 @@
+
+const today = new Date();
+
+function durationBetweenDates(dateStart = '24 Feb 2022', dateEnd = today.toString(), dimension = 'days') {
```
Це не так критично, але тепер дефолтний параметр функції залежить від зовнішньої змінної, що також не дуже ок. В ідеалі треба знайти рішення, при якому не буде як проблеми з типами даних,  так і залежностей від зовнішніх змінних.

daria-proleskovska committed:
Improvement

- task 1: remove 'today' variable
